packages/medusa/src/services/cart.ts

```diff
@@ -476,6 +490,25 @@ class CartService extends TransactionBaseService {
 
         const createdCart = cartRepo.create(rawCart)
         const cart = await cartRepo.save(createdCart)
+
+        if (
+          this.featureFlagRouter_.isFeatureEnabled([
+            SalesChannelFeatureFlag.key,
+            MedusaV2Flag.key,
+          ])
+        ) {
+          await this.remoteLink_.create({
+            cartService: {
+              cart_id: cart.id,
+            },
+            salesChannelService: {
+              sales_channel_id: (
+                await this.getValidatedSalesChannel(data.sales_channel_id)
```

Comment:
thought: I believe we've already done this to get the `rawCart.sales_channel_id`. Can we reuse?